File: lib/src/entity/user_entity.dart
import 'package:projeto_home_service/src/entity/address_entity.dart';

class UserEntity {
  int? userId;
  final String? name;
  final String? email;
  final String? password;
  final String? cpf;
  final AddressEntity? address;

  UserEntity({
    this.userId,
    required this.name,
    required this.cpf,
    required this.email,
    required this.password,
    this.address,
  });

  factory UserEntity.fromJson(Map<String, dynamic> json) {
    return UserEntity(
      userId: json['userId'],
      name: json['name'],
      email: json['email'] ?? '',
      password: json['password'],
      cpf: json['cpf'] ?? '',
      address: AddressEntity.fromJson(json['address'] ?? {}),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'name': name,
      'email': email,
      'password': password,
      'cpf': cpf,
      'address': address?.toJson(),
    };
  }
}

File: src/main/java/com/homeservice/api/entity/UserEntity.java
package com.homeservice.api.entity;

import java.io.Serializable;
import java.util.Objects;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "tb_user")
public class UserEntity implements Serializable{
	private static final long serialVersionUID = -3028711885595667424L;

	@Id				
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer userId;
	
	private String cpf;
	
	private String name;
	
	private String email;
		
	private String password;
	
	@OneToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "address_id", referencedColumnName = "addressId")
	private AddressEntity address;
	
	public UserEntity() {
		
	}
	
	public UserEntity(Integer userId, String name, String cpf, String email, String password, AddressEntity address) {
		this.userId = userId;
		this.name = name;
		this.email = email;
		this.password = password;
		this.cpf = cpf;
		this.address = address;
		
	}

	public Integer getUserId() {
		return userId;
	}

	public void setUserId(Integer userId) {
		this.userId = userId;
	}
	
	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getCpf() {
		return cpf;
	}

	public void setCpf(String cpf) {
		this.cpf = cpf;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}
	
	public AddressEntity getAddress() {
		return address;
	}

	public void setAddress(AddressEntity address) {
		this.address = address;
	}

	@Override
	public int hashCode() {
		return Objects.hash(email, name, password, userId);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		UserEntity other = (UserEntity) obj;
		return Objects.equals(email, other.email) && Objects.equals(name, other.name)
				&& Objects.equals(password, other.password) && Objects.equals(userId, other.userId);
	}

}

